microservices/download/index.js

```diff
@@ -3,14 +3,17 @@ const utils = require("./utils.js");
 const port = parseInt(process.env.PORT) || 8080;
 
 const server = http.createServer(async (req, res) => {
-  // Since req.url is our URL pluse query params, lets split them first.
-  const path = req.url.split("?");
+  // Since req.url is our URL plus query params, lets split them first.
+  const url = req.url.split("?");
+  const path = url[0];
+  const queryParams = url[1];
```

Comment:
If you want to keep these const definitions and otherwise meld my changes of `index.js` into this PR it should apply pretty cleanly in terms of like diff conflicts and such.

Either this way or my branch are more readable than before, IMO. Your version here might be even slightly nicer, it's pretty subjective tho. `path` and `queryParams` seems _very_ readable, IMO, as you have it on this branch already. 👍

Reply:
I ended up with this on my branch, which I wasn't in love with but couldn't think of how to get exactly the names I wanted and the readability I wanted:

```js
  let [urlPath, urlParams] = req.url.split("?");
```

The [destructuring assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment) on my branch, while concise, is IMO more confusing and less readable.

And not much point in reminding where these consts _came from_ (`urlPath` and `urlParams`) when `path` and `queryParams` are so descriptive, without fluff and readable.

So that's my thoughts why your variable names/assignment approach works better, IMO.